src/files/renameFile.api.js

```diff
@@ -1,41 +1,59 @@
 const buildApiHandler = require("../api-utils/build-api-handler");
 const filesService = require("./files.service");
 const paramsValidator = require("../middlewares/params-validator");
-const fileNameResolver = require("../middlewares/file-name-resolver");
-const userResolver = require("../middlewares/user-resolver")
+const fileUtils = require("./file.utils")
+const userResolver = require("../middlewares/user-resolver");
+const { logReceiver } = require("../logs/log-events");
+const config = require("../config");
 
 async function controller(req, res) {
-  const { id } = req.query;
-  const {newName, user} = req.body;
+  const { id } = req.params;
+  const { newName, user } = req.body;
 
-  const existingFile = await filesService.getFile(id, user);
+  const existingFile = await filesService.doesFileIdExistsForUser(id, user);
```

Comment:
directly call the db update api and use its return value to know if the file is successfully modified or not